base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -158,6 +160,12 @@ impl SenderTransactionInitializer {
         self
     }
 
+    /// Enable or disable spending of an amount less than the fee
+    pub fn with_fee_guard(&mut self, prevent_fee_gt_amount: bool) -> &mut Self {
```

Comment:
nit: Rather keep the name the same so that it's easy to find
```suggestion
    pub fn with_prevent_fee_gt_amount(&mut self, prevent_fee_gt_amount: bool) -> &mut Self {
```